src/Servers/Kestrel/Core/src/Internal/Http3/Http3Stream.cs

```diff
@@ -485,8 +487,16 @@ private Task ProcessUnknownFrameAsync()
                 _requestHeaderParsingState = RequestHeaderParsingState.Trailers;
             }
 
-            QPackDecoder.Decode(payload, handler: this);
-            QPackDecoder.Reset();
+            try
+            {
+                QPackDecoder.Decode(payload, handler: this);
+                QPackDecoder.Reset();
+            }
+            catch (QPackDecodingException ex)
```

Comment:
https://quicwg.org/base-drafts/draft-ietf-quic-http.html#header-size-constraints
> A server that receives a larger header section than it is willing to handle can send an HTTP 431 (Request Header Fields Too Large) status code

(HTTP/2) https://github.com/dotnet/aspnetcore/issues/17861

We would like to implement 431s eventually, they're easier to understand than InternalError. Do we get a specific enough error back from the decoder to do that?

At least file an HTTP/3 specific issue for this.

Reply:
Done - https://github.com/dotnet/aspnetcore/issues/33622